main/tests/server/src/com/google/refine/commands/CommandTestBase.java

```diff
@@ -40,4 +43,13 @@ public void setUpRequestResponse() {
     protected void assertCSRFCheckFailed() {
         TestUtils.assertEqualsAsJson(writer.toString(), "{\"code\":\"error\",\"message\":\"Missing or invalid csrf_token parameter\"}");
     }
+
+    /**
+     * Convenience method to check that CSRF protection was NOT triggered
+     */
+    protected void assertErrorNotCSRF() {
+        String response = writer.toString();
+        assertTrue(response.contains("\"code\":\"error\""));
```

Comment:
This relies on the response containing JSON formatted in a specific way (without spaces around `:`), which I'd rather avoid. What motivates you not to use the helper above?

Reply:
If "helper above" means `assertCSRFCheckFailed()`, it's testing for a different case. This helper is checking for `code: "error"` and `message: <anything but invalid csrf_token>`.

I can invest additional effort in coding it to protect against a potential future JSON formatting change if you think it's important.

Reply:
Ah right, I missed the negation. Yes I think it would be cleaner to parse the JSON and then access the fields of the resulting `JsonNode`.

Reply:
I've updated the test. I'm open to suggestions for a better method name or doc string to make its purpose clearer.